### PR TITLE
GithubEditorTest using wait.click, more stable selector in editor step pulldown

### PR DESCRIPTION
**Pull request by kshultzCB**
# GithubEditorTest updates, plus a more stable selector to target
This PR includes two main things.

1. Conversion to `wait.click()`, instead of `wait.until(things).click()`, to hopefully de-flake these tests. `wait.click` has which has retry behavior and an initial check for `elementIsClickable`, built into it. Inspired by [PR 1544](https://github.com/jenkinsci/blueocean-plugin/pull/1544).
2. Addition of `data-functionName={step.functionName}` in the pulldown for creating a step in the editor. This, hopefully, provides a more stable selector for the tests to look for. It matches up nicely to the step which will be created in the Jenkinsfile - e.g., "Shell Script" matches up to `sh`.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate: N/A.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests: this is an acceptance test addition. Unit and acceptance tests not applicable
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# To run:

- Follow the instructions provided in [the acceptance tests' README](https://github.com/kshultzCB/blueocean-plugin/tree/add-parallel-editor-test/acceptance-tests) for setting up the "live" acceptance tests
- Invoke `mvn test -Dprofile=live -Dtest=GithubEditorTest`

### CODE REVIEW

File: blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class AddStepSelectionSheet extends Component<DefaultProps, Props, State>
                 </div>
                 <div className="editor-step-selector">
                 {stepMetadata && stepMetadata.filter(isStepValidForSelectionUI).filter(this.state.searchFilter).sort(stepSorter).map(step =>
-                    <div tabIndex="0" onKeyPress={e => this.selectItemByKeyPress(e, step)}
+                    <div tabIndex="0" data-functionName={step.functionName} onKeyPress={e => this.selectItemByKeyPress(e, step)}
```
Comment (michaelneale) [inline blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx]: nice to see this pattern returning

Comment (michaelneale) [reply blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx]: although should it use data-function-name vs hybrid case? (maybe @sophistifunk has a style opinion)

**Comment by michaelneale** [comment]
nice one :bee:

File: blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx (hunk 1 above)
Comment (sophistifunk) [reply blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx]: `data-` should probably be viewed as a namespace, so what you put after it should be dictated by how you need to used it, and we don't want hyphens in our object prop names when we can avoid it.

Comment (sophistifunk) [reply blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx]: TL/DR: `data-functionName` is good. That's what I'd do. Therefore it must be good, of course. ∎

Comment (michaelneale) [reply blueocean-pipeline-editor/src/main/js/components/editor/AddStepSelectionSheet.jsx]: ah good to know @sophistifunk 
image

**Pull request merged**